### DISCUSSION TITLE
Blueprint route changes mimetype from 'multipart/form-data' to 'application/x-www-form-urlencoded'

### DISCUSSION THREAD

**Discussion by automata** [pallets/flask#5097] [Q&A]
No sure if a bug, but I couldn't find anywhere else the reason for this behaviour of Blueprints.

When send the same `multipart/form-data` request to `/foo` and `/bar` bellow, I get different mimetypes:

```python
# app.py

from flask import Flask, request
from foo_bp import foo_bp

app = Flask(__name__)

app.register_blueprint(foo_bp)

@app.route("/bar", methods=["POST"])
def bar():
    print(request.mimetype) # =>  multipart/form-data
    return {}, 200
```

```python
# foo_bp.py

from flask import Blueprint, request

foo_bp = Blueprint("foo_bp", __name__, url_prefix="/foo")

@foo_bp.route("/", methods=["POST"])
def foo():
    print(request.mimetype) # => application/x-www-form-urlencoded
    request {}, 200
```

I'm using `asyncio` to send the requests.

For `/foo` (a Blueprint) I get `application/x-www-form-urlencoded` while for `/bar` I get the expected `multipart/form-data`.

Also, `request.files` in the Blueprint has an empty list.

Am I missing something? Thank you in advance.

Environment:

- Python version: 3.10.11
- Flask version: 2.2.3

**Comment by davidism** [comment]
I can't reproduce this issue with the information provided. Posting `multipart/form-data` to a route sets `request.mimetype` to `multipart/form-data`, blueprints do not affect that. When asking a question, be sure to include a minimal reproducible example.

**Reply by automata** [reply]
Sorry about that, I updated the original code snippets for a working reproducible example. While testing the example, I noticed that the Blueprint was answering with `308` instead of `200` and searching for that I found a solution (https://github.com/pallets/flask/issues/3271): adding `strict_slashes=False` to the Blueprint route solved the problem, making it return `200` as expected and with the right `request.mimetype`.

Here is the working example, with the fix:

```python
# app.py

from flask import Flask, request
from foo_bp import foo_bp

app = Flask(__name__)

app.register_blueprint(foo_bp)

@app.route("/bar", methods=["POST"])
def bar():
    print(request.mimetype) # =>  multipart/form-data
    return {}, 200
```

```python
# foo_bp.py

from flask import Blueprint, request

foo_bp = Blueprint("foo_bp", __name__, url_prefix="/foo")

@foo_bp.route("/", methods=["POST"], strict_slashes=False)
def foo():
    print(request.mimetype) # => multipart/form-data
    request {}, 200
```

**Reply by davidism** [reply]
This still doesn't make sense. A 308 redirect preserves the request body and content type. It's unlikely disabling `strict_slashes` is actually what you want. It seems like you have some other issue and have misidentified Flask as the cause.

**Reply by automata** [reply]
Interesting, let me try to share the `sender.py` script as well, I believe it's the only missing bit for a working example:

```python
# sender.py

import os
import aiohttp
import asyncio

def get_asset_files(asset_dir):
    asset_files = []
    for filename in os.listdir(asset_dir):
        asset_files.append((
            filename,
            os.path.join(asset_dir, filename),
            "text/plain"))
    return asset_files

def create_form_data(files, key_value_data):
    data = aiohttp.FormData()
    for filename, filepath, mimetype in files:
        data.add_field(
            filename,
            open(filepath, 'rb'),
            filename=filename,
            content_type=mimetype)
    for key, value in key_value_data.items():
        data.add_field(key, value)
    return data

async def main():
    asset_dir = "./asset"
    asset_data = {
        "some_key": "some_value"
    }
    asset_files = get_asset_files(asset_dir)
    data = create_form_data(asset_files, asset_data)
    timeout = aiohttp.ClientTimeout(total=600)
    async with aiohttp.ClientSession(timeout=timeout) as session:
        res_json = await session.post("http://localhost:5000/foo", data=data)
        print(res_json)

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```
After checking https://github.com/aio-libs/aiohttp/issues/5577 I started suspecting of aiohttp.

**Reply by davidism** [reply]
Sounds like that client isn't handling 307 and 308 redirects properly. That's a bug with that library, not with Flask.

**Reply by automata** [reply]
Exactly, it seems like it, thank you so much!
